certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -225,6 +238,32 @@ def updated_mod_ssl_conf_digest(self):
         """Full absolute path to digest of updated SSL configuration file."""
         return os.path.join(self.config.config_dir, constants.UPDATED_MOD_SSL_CONF_DIGEST)
 
+    def openssl_version(self):
+        """Lazily retrieve openssl version"""
+        if self._openssl_version:
+            return self._openssl_version
+        # Step 1. Check for LoadModule directive
+        try:
+            ssl_module_location = self.parser.modules['ssl_module']
```

Comment:
Today I learned that this path can also be relative to the Apache server root. See https://httpd.apache.org/docs/2.4/mod/mod_so.html#loadmodule.

I noticed this because I ran our Apache test farm tests and they failed on Fedora trying to open this file because a relative path is used. (The good news is the tests on all other systems in `apache2_targets.yaml` used the correct file though!)

Reply:
which test exactly was this?

Reply:
I'm guessing this one (running now): https://travis-ci.com/github/certbot/certbot/jobs/297589213

Reply:
[all of the test farm tests](https://travis-ci.com/github/certbot/certbot/builds/153094830) are passing now, so unless it was a different test that you ran manually, looks like it's good now.